### DISCUSSION TITLE
Creating an update (create-once-only) form type without datasource?

### DISCUSSION THREAD

**Discussion by philsonchoi** [Budibase/budibase#9214] [Help] [answered]
I was wondering if there is a way to create a form for user's personal details? In the sense that the user would create a row once, and only update that specific row afterwards. Using "Update" instead of "Create" shows the error of no data to display.

I used the "Customer Account Portal" template as a reference, but the template's "Edit my details" (/customer-details/:id
) screen shows the error "We couldn't find a row to display" both in the preview mode and in the builder. Additionally, the methods of creating new rows seems to be outdated to the side panel methods.

![Screenshot 2023-01-04 at 5 09 54 PM](https://user-images.githubusercontent.com/73387343/210522228-4d33bff4-4d9f-456a-a777-7b0910d90617.png)

Template link: https://budibase.com/portals/templates/customer-account-portal/

Any advice or sample templates for how this could be implemented would be greatly appreciated.

**Comment by melohagan** [comment, marked as answer]
Hey @philsonchoi 

You can add custom columns to the users' table: https://docs.budibase.com/docs/users-table#custom-columns

Then if you add a Form Block, you can do something like so to 'edit my details':
![Screenshot 2023-01-04 at 13 46 15](https://user-images.githubusercontent.com/101575380/210568525-1e5dcc02-8101-48e3-ad42-d60e7612cbec.png)
